Issue #1701: Events page displays year for events scheduled to start or end at a future year

alvindera97 opened the pull request:
This is the PR replacing https://github.com/python/pythondotorg/pull/1713. The functional tests have been removed and logical/functional concerns have been replicated with unit tests that don't require selenium to run. 

_This is basically because of the CI Pipeline which makes no provision for installing selenium's web driver required by selenium to run the [headless] functional tests._ 

**There is something that I would like to note:**
— When I was running the unit tests, I found out that Django was somehow caching the results of the tests and so the subsequent test results I was getting were rather untrustworthy. 

• I was able to get around this by adding a `tearDown()` method to the relevant unit test class. This `tearDown()` method basically clears the cache at the end of each test (`cache.clear()`) suggested [here](https://www.reddit.com/r/django/comments/6fx18f/psa_django_core_caching_pollutes_test_results/?utm_source=share&utm_medium=web2x&context=3)

Please feel free to ask any questions. This is basically my first shot at contributing to open source software.

alvindera97 committed:
Add Methods To Check If Event Starts And Ends This Year

alvindera97 committed:
Add Methods To Check If Event Starts And Ends This Year

alvindera97 committed:
Add Methods To Check If Event Starts And Ends This Year

alvindera97 committed:
Set Up Templates For Querying Start And End Years

Passed variables to the time_tag template [time_tag.html] that checks if an event was scheduled to start or end with the current year.

alvindera97 committed:
Set Up Templates For Querying Start And End Years

Passed variables to the time_tag template [time_tag.html] that checks if an event was scheduled to start or end with the current year.

alvindera97 committed:
Set Up Templates For Querying Start And End Years

Passed variables to the time_tag template [time_tag.html] that checks if an event was scheduled to start or end with the current year.

alvindera97 committed:
Insert New Test Data And Update test_views.py

More events are created to test particular scenarios of events especially events set to start or end at a future year.

alvindera97 committed:
Insert New Test Data And Update test_views.py

More events are created to test particular scenarios of events especially events set to start or end at a future year.

alvindera97 committed:
Insert New Test Data And Update test_views.py

More events are created to test particular scenarios of events especially events set to start or end at a future year.

alvindera97 committed:
Time Tag Now Shows Year For Events With Details Not Within The Current Year

The time tag now displays the year when an event will occur. This is only for events that have been scheduled to start or end in at a future year. The accompanying functional tests have also been included.

alvindera97 committed:
Time Tag Now Shows Year For Events With Details Not Within The Current Year

The time tag now displays the year when an event will occur. This is only for events that have been scheduled to start or end in at a future year. The accompanying functional tests have also been included.

alvindera97 committed:
Time Tag Now Shows Year For Events With Details Not Within The Current Year

The time tag now displays the year when an event will occur. This is only for events that have been scheduled to start or end in at a future year. The accompanying functional tests have also been included.

alvindera97 committed:
Time Tag Now Shows Year For Events With Details Not Within The Current Year

The time tag now displays the year when an event will occur. This is only for events that have been scheduled to start or end in at a future year. The accompanying functional tests have also been included.

alvindera97 committed:
Move All Test Data To Functional Test

All test data concerning the provision of data to serve the functional tests have been moved to the functional test.
As it improves readability.
All other test data at test_views.py was reset
to accommodate for the reduction in number of test data instances.

alvindera97 committed:
Move All Test Data To Functional Test

All test data concerning the provision of data to serve the functional tests have been moved to the functional test.
As it improves readability.
All other test data at test_views.py was reset
to accommodate for the reduction in number of test data instances.

alvindera97 committed:
Move All Test Data To Functional Test

All test data concerning the provision of data to serve the functional tests have been moved to the functional test.
As it improves readability.
All other test data at test_views.py was reset
to accommodate for the reduction in number of test data instances.

alvindera97 committed:
Move All Test Data To Functional Test

All test data concerning the provision of data to serve the functional tests have been moved to the functional test.
As it improves readability.
All other test data at test_views.py was reset
to accommodate for the reduction in number of test data instances.

alvindera97 committed:
Functional Test For Displaying Year Of Event For Future Events Now Implemented With Unit Tests

alvindera97 committed:
Functional Test For Displaying Year Of Event For Future Events Now Implemented With Unit Tests

alvindera97 committed:
Functional Test For Displaying Year Of Event For Future Events Now Implemented With Unit Tests

alvindera97 committed:
Functional Test For Displaying Year Of Event For Future Events Now Implemented With Unit Tests

alvindera97 committed:
Remove Functional Test For Displaying Year Of Future Event

Since the current CI at the main branch does not support selenium [web driver] operations, the functional test which depends on selenium to run has been removed.

alvindera97 committed:
Remove Functional Test For Displaying Year Of Future Event

Since the current CI at the main branch does not support selenium [web driver] operations, the functional test which depends on selenium to run has been removed.

alvindera97 committed:
Remove Functional Test For Displaying Year Of Future Event

Since the current CI at the main branch does not support selenium [web driver] operations, the functional test which depends on selenium to run has been removed.

alvindera97 committed:
Remove Functional Test For Displaying Year Of Future Event

Since the current CI at the main branch does not support selenium [web driver] operations, the functional test which depends on selenium to run has been removed.

ferdnyc commented on .gitignore:
```diff
@@ -17,3 +17,8 @@ static/stylesheets/no-mq.css
+
+# Ignore Microsoft Visual Studio Code's and Jetbrain's Pycharm Files
+.vscode/
+.idea/
```
This seems kind of unnecessary, as the repository contains no `.vscode` or `.idea` directories, nor does anything in the repo generate such files.

Presumably those are added because you use tools that generate them locally. But if the project's `.gitignore` is going to contain exclusions for your personal metadata files from your personal choice of editor(s), then isn't it going to end up having to contain exclusions for **every** possible tool metadata file? No matter what it is, no matter how obscure, guarantee someone somewhere is generating it in their clone right now.

Fortunately, Git has a built-in solution for this. If you want to protect against accidentally adding files/directories that you generate locally,  when creating  your **local** commits, you should instead add those exclusions to the file `.git/info/exclude`. That's where local additions to the ignore list belong.

The Git developers have unfortunately not been very successful at disseminating this information, though it is presented right at the top of the [`gitignore{5}`](https://man7.org/linux/man-pages/man5/gitignore.5.html) man page.

(The problem is, `gitignore(5)` doesn't explain **why** some things belong in `.gitignore` and other things belong in `.git\info\exclude`. `.gitignore` gets a whole paragraph of explanation, and then it's followed by  a bullet point: "Patterns read from `$GIT_DIR/info/exclude`." That's it, end of item.)

So, what I'm saying is: Not your fault, extremely common mistake. But, `.gitignore` is still the wrong place for such additions. It's meant for (quoting from `gitignore(5)`) "patterns for files generated as part of the project build.".

alvindera97 replied:
Good day, 

Thanks for getting back to me on this PR. I have updated ..gitignore file and I have made the changes as you've instructed. Please do reach out if you have any questions or clarifications.

alvindera97 committed:
Handle Case When Call To Next Event Returns None

alvindera97 committed:
Handle Case When Call To Next Event Returns None

alvindera97 committed:
Handle Case When Call To Next Event Returns None

alvindera97 committed:
Update Package Versions For Dev Requirements Required For Testing

alvindera97 committed:
Update Package Versions For Dev Requirements Required For Testing

alvindera97 committed:
Merge branch 'main' into change-functional-test-to-unit-test

ewdurbin committed:
add snippet for google analytics 4

eshaanmandal committed:
Fixed Issue#2190 | Homepage code syntax error (#2191)

* Fixed Issue#2190Now the Homepage code sample doesn't have syntax error

* Another minor typo in the next line

Co-authored-by: Eshaan <eshaan@pop-os.localdomain>

ewdurbin committed:
add a sponsorship export function via admin (#2231)

renatooliveira committed:
Add management command to create sponsor vouchers for PyCon 2023 (#2233)

* ignore Makefile .state folder

* add test and docker_shell command to Makefile

* Add command to create pycon vouchers for sponsors

* Update sponsors/management/commands/create_pycon_vouchers_for_sponsors.py

* Update sponsors/management/commands/create_pycon_vouchers_for_sponsors.py

Co-authored-by: Ee Durbin <ewdurbin@gmail.com>

ewdurbin committed:
updates to sponsor contract template (#2237)

closes #2235
closes #2236

ewdurbin committed:
update prefix for additional discounted registration vouchers

di committed:
Add missing migrations (#2246)

di committed:
Add support for Sigstore bundle files (#2247)

di committed:
Add a check in CI for missing migrations (#2248)

ewdurbin committed:
sponsorship: allow admins to temporarily unlock finalized sponsorships for editing (#2249)

ewdurbin committed:
bump rate limit for api

ewdurbin committed:
sponsorship form: fix rendering of application form

When benefits are configured for packages that are to the right of
packages that *do not* have a benefit, we should correctly render.

This was effectively "gravity left" for the icons on the application.

ewdurbin committed:
enable customizing admin (#2263)

psf staff spend a lot of time between us.pycon.org and python.org django admin
this enables configuration of a different theme for python.org so its easier
to differentiate. thanks to @swiencks for pointing out how easy to confuse they are.

ewdurbin committed:
revert inadvertently committed.

was headed this direction while responding to https://status.python.org/incidents/kwsfwt5q4rg1

ewdurbin committed:
add notes regarding basic membership (#2271)

ewdurbin committed:
missed

ewdurbin committed:
fix test

ewdurbin committed:
handle malformed events from ics calendars (#2281)

ref: https://python-software-foundation.sentry.io/issues/4210332341/events/859bc65a91524856a24f6dae19f52d4b/

ewdurbin committed:
allow acceptance of nominations after deadline

luzfcb committed:
Use "static" templatetag insteadof the deprecated "admin_static" (#2288)

luzfcb committed:
Convert custom_storages.py to a python module (#2289)

* Convert custom_storages.py to a python module

* Do not use a mutable object as the default value of the function argument

* Install production dependencies on GitHub action to running the tests

* Avoid upgrading Django by defining versions to production deps

Whitenoise > 6.0.0 and django-storages > 1.12.3 requires Django 3.2 or higher

* Remove imports on custom_storages __init__ file to avoid error when execute the tests

ewdurbin committed:
no longer sponsored (#2296)

ewdurbin committed:
Update feature_request.md

mlcohen committed:
Update python.org socialize menu to include a link to the PSF Mastodon account (#2323)

* Update python.org socialize menu to include a link to the PSF Mastodon account

Updated socialize menu to include a link to the official PSF Mastodon account.

Details:

- Added Mastodon icon to Pythonicon font family. Used IcoMoon to generate icons
- Updated all related CSS and Sass files
- Updated the socialize menu in the base.html template

To see all rendered Pythonicon fonts, open `static/fonts/demo.html`

* Updated help and close icons unicodes

Help icon is now assigned unicode `\3f` (`?`). Close icon is now assigned unicode `\58` (`X`). Required regenerating fonts and updating CSS + Sass files.

ewdurbin committed:
upgrade elasticsearch client

ewdurbin committed:
upgrade django-haystack

ewdurbin committed:
use new backend

ewdurbin committed:
make index name configurable

ewdurbin committed:
this is just a 1/2/3 differentiator, disregard in search

hugovk committed:
Add config file for Read the Docs (#2328)

* Add config file for Read the Docs

* Remove comment

hugovk committed:
Remove broken Twitter widget (#2329)

* Remove broken Twitter widget

* Remove broken Twitter widget

ewdurbin committed:
fix misleading link in site-tree fixture

ewdurbin committed:
Update sponsor_new_application.txt

ewdurbin committed:
update sponsor contract with new info, fix sponsorship year

ewdurbin committed:
fix notice in post-roll on sponsor application

ewdurbin committed:
filtering by current_year where necessary

ewdurbin committed:
fix deletion of BenefitFeatures

ewdurbin committed:
missing migration

ewdurbin committed:
missing migration

jessiebelle committed:
Sponsorship - adding renewal option for contract generation (#2344)

* WIP renewal work

* test fix

* removing venv added files

* tidy and fixup

* test fixup after logic changes

* Sponsorship renewal review (#2345)

* add rewnewal to the admin view for sponsorship

* use the sponsorship form directly rather than editing template

* include previous effective date in context/review form for renewals

* update to real renewal contract

* missing migration

---------

Co-authored-by: Ee Durbin <ewdurbin@gmail.com>

ewdurbin committed:
formatting tweaks to the renewal contract

ewdurbin committed:
actually add template change

varunhc committed:
updated python issue tracker and pep links (#2318)

* updated python issue tracker and pep links

* Update templates/base.html

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

---------

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

jessiebelle committed:
Fix for SVG upload in sponsorship application (#2352)

* changing imagefield to file field on sponsorupdateform and sponsorshipapplicationform. adding django file validator to these two and the sponsor model

* allow png upload

* adding test to assert that svg can be uploaded

* fix migration

ewdurbin committed:
update sponsor voucher fulfillment for 2024

jessiebelle committed:
add new incorporation fields to form and sponsor model (#2351)

* adding new fields to form and sponsor model

* adding new optional fields to update sponsor application form

* fix migrations

* modifying existing tests to assert new fields are correctly saved

* Re-structure and re-order sponsor information form

* update and refactor of new_sponsorship_application_form to be usable for editing

* remove migration

* new migration file added

* merge

* add migration

* fix tests to reflect update in template

---------

Co-authored-by: Ee Durbin <ewdurbin@gmail.com>

jessiebelle committed:
2342 display only benefits associated with the year of the sponsorship package being edited (#2356)

* WIP

* WIP

* WIP

* we finally got there with the qs filter

* removing redundant tests

* removing unused imports

* missed one

sethmlarson committed:
Add support for hosting SPDX-2 SBOMs alongside release artifacts (#2359)

sethmlarson committed:
Link to SBOM user documentation from release detail page (#2363)

ewdurbin committed:
LinkedIn stuff (#2367)

* Adds linkedin to socialize
* Adds linkedin to sponsor profile

Closes #2365

joncle committed:
Update job_detail.html (#2298)

Explicity escape the job description for the meta og:description where the description contains html which can throw off the rendering of the page...

See https://www.python.org/jobs/7314/ for example.

cfsok committed:
Add DearPyGui to home page (#2362)

noelleleigh committed:
`download:download_release_detail` view: Display file sizes with human-readable units (#2354)

Make the release file sizes more readable by passing them through the
[filesizeformat filter][1].

[1]: https://docs.djangoproject.com/en/2.2/ref/templates/builtins/#filesizeformat

ewdurbin committed:
Move to pandoc for rendering sponsorship contracts (#2343)

* Move to pandoc for rendering sponsorship contracts

* use renewal template if applicable

* consistency

* formatting tweaks

* bit cleaner names for files

* grinding out formatting

hugovk committed:
Prioritise 64-bit downloads over 32-bit (#2311)

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

dorian-adams committed:
Redirect legacy community page (#2319)

Use regular expression to redirect all legacy 'community-landing/' paths, including any children.

dependabot[bot] committed:
Bump beautifulsoup4 from 4.9.3 to 4.11.2 (#2239)

Bumps [beautifulsoup4](https://www.crummy.com/software/BeautifulSoup/bs4/) from 4.9.3 to 4.11.2.

---
updated-dependencies:
- dependency-name: beautifulsoup4
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump lxml from 4.6.3 to 4.9.2 (#2212)

Bumps [lxml](https://github.com/lxml/lxml) from 4.6.3 to 4.9.2.
- [Release notes](https://github.com/lxml/lxml/releases)
- [Changelog](https://github.com/lxml/lxml/blob/master/CHANGES.txt)
- [Commits](https://github.com/lxml/lxml/compare/lxml-4.6.3...lxml-4.9.2)

---
updated-dependencies:
- dependency-name: lxml
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ambv committed:
Revert "Move to pandoc for rendering sponsorship contracts (#2343)" (#2374)

This reverts commit a4990b711ccd6e34e87d8d4b97169846fba5414f.

ambv committed:
Remove stranded </span> from templates/base.html (#2377)

This was introduced in #2367.

ewdurbin committed:
Redirect cleanup (#2378)

* Revert "Redirect legacy community page (#2319)"

This reverts commit f88599b49d6514168f45920a531823e22d7570d6.

* move community-landing redirect to nginx

* move remainging RedirectViews to nginx

* match intended recirect from community-landing pr

ewdurbin committed:
fix download redirects to match previous behavior (#2393)

ambv committed:
Move to pandoc for rendering sponsorship contracts (take 2) (#2375)

* Move to pandoc for rendering sponsorship contracts

* use renewal template if applicable

* consistency

* formatting tweaks

* bit cleaner names for files

* grinding out formatting

* update sig block to indent

* remove debug output from page break pandoc filter

* update template with formatting tweaks

- intcomma sponsorship fee
- brute force indent notices
- period after all seciton headings

* consolidate templates

---------

Co-authored-by: Ee Durbin <ewdurbin@gmail.com>

ewdurbin committed:
implement a static URL for editing sponsor details (#2394)

this is useful for our sponsorship cooridnator to include in comms

ewdurbin committed:
add celery (#2395)

ewdurbin committed:
Deploy to Cabotage (#2396)

* add health check endpoint

* add Dockerfile for cabotage

* bind to cabotage socket

* rename config

* SOURCE_VERSION -> SOURCE_COMMIT

* migrate from nginx buildpack

* collectstatic

* fixup docker files

* add ca cert verification to haystack/elasticsearch

* run worker and worker-beat

ewdurbin committed:
Redirect Code of Conduct documents to polices.python.org (#2399)

ambv committed:
Update Django to 2.2.28 (#2398)

ewdurbin committed:
followup on 2399 (#2400)

- Repair syntax for redirects
- Remove now extraneous redirect

ewdurbin committed:
unidecode filenames for sponsorship contract export

hugovk committed:
Update python3statement URL (#2433)

cclauss committed:
Update ci.yml (#2424)

This could have been automated in
* #2401 

* https://github.com/actions/checkout/releases
* https://github.com/actions/setup-python/releases
* https://github.com/actions/cache/releases

Co-authored-by: Łukasz Langa <lukasz@langa.pl>

hugovk committed:
Docs: Fix links (#2412)

* Fix link to licence

* Update redirected links

* Fix link to GitHub docs

* Replace Travis CI with GitHub Actions

---------

Co-authored-by: Łukasz Langa <lukasz@langa.pl>

dependabot-preview[bot] committed:
Bump factory-boy from 2.9.2 to 3.2.0 (#1715)

Bumps [factory-boy](https://github.com/FactoryBoy/factory_boy) from 2.9.2 to 3.2.0.
- [Release notes](https://github.com/FactoryBoy/factory_boy/releases)
- [Changelog](https://github.com/FactoryBoy/factory_boy/blob/master/docs/changelog.rst)
- [Commits](https://github.com/FactoryBoy/factory_boy/compare/2.9.2...3.2.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>
Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>
Co-authored-by: Łukasz Langa <lukasz@langa.pl>

dependabot[bot] committed:
Bump gunicorn from 19.9.0 to 21.2.0 (#2384)

Bumps [gunicorn](https://github.com/benoitc/gunicorn) from 19.9.0 to 21.2.0.
- [Release notes](https://github.com/benoitc/gunicorn/releases)
- [Commits](https://github.com/benoitc/gunicorn/compare/19.9.0...21.2.0)

---
updated-dependencies:
- dependency-name: gunicorn
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cclauss committed:
Keep GitHub Actions up to date with GitHub's Dependabot (#2401)

* Keep GitHub Actions up to date with GitHub's Dependabot

* https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot
* https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

* Apply suggestions from code review

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

* Update .github/dependabot.yml

---------

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>
Co-authored-by: Ee Durbin <ewdurbin@gmail.com>

ewdurbin committed:
allow clearing textbenefits via admin

ewdurbin committed:
add plausible.io analytics

ewdurbin committed:
upgrade django-allauth to bypass setuptools 70.0.0 issue

dependabot[bot] committed:
Bump gunicorn from 19.9.0 to 22.0.0 (#2436)

Bumps [gunicorn](https://github.com/benoitc/gunicorn) from 19.9.0 to 22.0.0.
- [Release notes](https://github.com/benoitc/gunicorn/releases)
- [Commits](https://github.com/benoitc/gunicorn/compare/19.9.0...22.0.0)

---
updated-dependencies:
- dependency-name: gunicorn
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump django-sitetree from 1.17.0 to 1.18.0 (#2406)

Bumps [django-sitetree](https://github.com/idlesign/django-sitetree) from 1.17.0 to 1.18.0.
- [Changelog](https://github.com/idlesign/django-sitetree/blob/master/CHANGELOG)
- [Commits](https://github.com/idlesign/django-sitetree/compare/v1.17.0...v1.18.0)

---
updated-dependencies:
- dependency-name: django-sitetree
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump lxml from 4.9.2 to 5.2.2 (#2450)

Bumps [lxml](https://github.com/lxml/lxml) from 4.9.2 to 5.2.2.
- [Release notes](https://github.com/lxml/lxml/releases)
- [Changelog](https://github.com/lxml/lxml/blob/master/CHANGES.txt)
- [Commits](https://github.com/lxml/lxml/compare/lxml-4.9.2...lxml-5.2.2)

---
updated-dependencies:
- dependency-name: lxml
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alvindera97 committed:
Merge remote-tracking branch 'origin/change-functional-test-to-unit-test' into successfully_rebased_cfttut

alvindera97 committed:
Merge branch 'change-functional-test-to-unit-test' of github.com:alvindera97/pythondotorg into successfully_rebased_cfttut
